packages/astro/src/core/dev/index.ts

```diff
@@ -347,7 +362,11 @@ export class AstroDevServer {
     }
     // if not found, fall back to default template
     else {
-      html = errorTemplate({ statusCode, title: 'Not found', tabTitle: '404: Not Found', message: pathname });
+      if(pathname === '/' && !pathname.startsWith(this.pathname)) {
+        html = subpathNotUsedTemplate(this.pathname);
+      } else {
+        html = errorTemplate({ statusCode, title: 'Not found', tabTitle: '404: Not Found', message: pathname });
```

Comment:
Slight suggestion: you could probably add a new conditional param to the 4xx template to conditionally show this message or not (also happy to pair on this if needed)